pkg/config/config.go

```diff
@@ -1261,6 +1285,9 @@ func (c *Config) FindHelperBinary(name string, searchPATH bool) (string, error)
 		return exec.LookPath(name)
 	}
 	configHint := "To resolve this error, set the helper_binaries_dir key in the `[engine]` section of containers.conf to the directory containing your helper binaries."
+	if bindirErr != nil {
+		configHint = fmt.Sprintf("Error looking up path relative to %s: [%s]  %s", bindirPrefix, bindirErr, configHint)
```

Comment:
Should this be a logrus.Warnf instead of a print?

Reply:
Oh yes indeed logging is much nicer!

By not having to track the error result, I ended up refactoring the code considerably, moving `bindirPath` calculation to a standalone function.

In the new code, I also track whether `os.Executable()` (or resolving symlinks from it) ever failed, in order to only log about it once.

I hope you still like how the new code looks, let me know if you'd like me to make any changes to it.